packages/i18n/src/index.tsx

```diff
@@ -157,10 +158,98 @@ export const createI18nContext = (
      */
     dict: (lang: string) => deepReadObject(dict, lang)
   };
-  return [translate, actions];
+  return [translate, actions as any];
 };
 
 type I18nContextInterface = ReturnType<typeof createI18nContext>;
 
 export const I18nContext = createContext<I18nContextInterface>({} as I18nContextInterface);
 export const useI18n = () => useContext(I18nContext);
+
+// -------------------------- Chained I18n -------------------
+type I18nFormatArgs = Record<string, string | number>;
+
+export type I18nPath<T> = {
+  [K in keyof T]: T[K] extends Record<string, unknown>
+    ? I18nPath<T[K]>
+    : T[K] extends (...args: any) => string
+    ? (...args: Parameters<T[K]>) => string
+    : T[K] extends string
+    ? (args?: I18nFormatArgs) => string
+    : never;
+};
+
+const buildI18nChain = <T,>(obj: T): I18nPath<T> => {
```

Comment:
what does the `,` after the generic does? I guess nothing but I saw it in a couple of places already so you never know 😅
```suggestion
const buildI18nChain = <T>(obj: T): I18nPath<T> => {
```

Reply:
In TSX generics need a comma after them. I'm not sure why, but the linters aren't happy with it unless there is one.

Reply:
Ah I know now
It's because of the `tsx` extension - without it `<T>` is interpreted as JSX

Reply:
That makes sense!